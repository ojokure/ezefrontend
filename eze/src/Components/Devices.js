import React from "react";
import DeviceCard from "./DeviceCard";
import Sidebar from "./Sidebar";
import device_images from "./imageArray";
import { StyledMarketplace } from "../Styles/StyledMarketPlace";
import Mainlayout from "./Mainlayout";

console.log(device_images);

function Devices(props) {
  const { devices, searchPriceCallback } = props;

  return (
    <StyledMarketplace>
      {/* <div className="categories"> Categories </div> */}
      <Sidebar searchPrice={searchPriceCallback} />
      {/* <Mainlayout /> */}

      <div className="devices">
        {devices?.map((device) => (
          <DeviceCard
            //  key={device.id}
            name={device.name}
            price={device.price}
            condition={device.condition}
            images={device_images}
          />
        ))}
      </div>
    </StyledMarketplace>
  );
}

export default Devices;
