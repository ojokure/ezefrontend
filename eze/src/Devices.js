import React from "react";
import styled from "styled-components";
import DeviceCard from "./DeviceCard";
import device_images from "./image_array";
import devices from "./utils/devices";

console.log(device_images);

const StyledMarketplace = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-content:center;
  width: 100%;
  margin-top: 5rem;

  .categories {
    display: flex;
    justify-content: center;
    /* align-items: center; */
    width: 16rem;
    height: 80rem;
    padding-right: 1.5rem;
    margin-left: 1.5rem;
    margin-right: 0.5rem;
    color: white;
    text-align: center;
    background: #1e242f;
    margin-top: 0.5rem;

    /* background .keyword {
      margin-top: 1rem;
    } */

    /* @media ${devices.mobile} {
      flex-direction: column;
      align-items: center;
    } */
  }

  .devices {
    display: flex;
    flex-wrap: wrap;
    width: 70%;
    justify-content: space-between;
    color: white;
    margin-right: 1.5rem;

  }
`;

function Devices(props) {
  const { devices } = props;

  return (
    <StyledMarketplace>
      <div className="categories"> Categories </div>
      <div className="devices">
        {devices?.map((device) => (
          <DeviceCard
            //  key={device.id}
            name={device.name}
            condition={device.condition}
            images={device_images}
          />
        ))}
      </div>
    </StyledMarketplace>
  );
}

export default Devices;
