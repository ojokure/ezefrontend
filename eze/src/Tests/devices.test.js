import React from "react";
import ReactDOM from "react-dom";
import Devices from "../Components/Devices";

import renderer from "react-test-renderer";

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<Devices></Devices>, div);
});

it("matches snapshot", () => {
  const tree = renderer.create(<Devices></Devices>).toJSON();
  expect(tree).toMatchSnapshot();
});
