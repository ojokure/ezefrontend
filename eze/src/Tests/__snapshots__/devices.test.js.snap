// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="sc-AxgMl Meypm"
>
  <div
    className="sc-AxhUy fAIeOp"
  >
    <div
      className="title-bar"
    >
      Categories
    </div>
    <div
      className="side-bar-menu"
    >
      <div
        className="all"
      >
         All 
      </div>
      <div
        className="bar-menu"
      >
         Iphone 
      </div>
      <div
        className="bar-menu"
      >
         Samsung 
      </div>
      <div
        className="bar-menu"
      >
         Ipad 
      </div>
      <div
        className="bar-menu"
      >
         MacBook 
      </div>
      <div
        className="bar-menu"
      >
         Airpods 
      </div>
      <div
        className="title-bar"
      >
         Price Filter 
      </div>
      <div>
        <input
          className="ant-input"
          id="search"
          name="min"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          placeholder="Min"
          type="text"
          value={0}
        />
      </div>
      <div>
         | 
      </div>
      <div>
        <input
          className="ant-input"
          id="search"
          name="max"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          placeholder="Max"
          type="text"
          value={10000}
        />
      </div>
      <div
        className="title-bar"
      >
         Storage 
      </div>
      <div
        className="bar-menu"
      >
         32gb 
      </div>
      <div
        className="bar-menu"
      >
         64gb 
      </div>
      <div
        className="bar-menu"
      >
         128gb 
      </div>
      <div
        className="bar-menu"
      >
         256gb 
      </div>
    </div>
  </div>
  <div
    className="devices"
  />
</div>
`;
